### PR TITLE
filter inline role policies for roles created by AWS

### LINKED ISSUE #331: AWSServiceRoleForAmazonGuardDuty should be excluded
Using AWS Nuke v2.7.0, we encountered this error:

`global - IAMRolePolicy - AWSServiceRoleForAmazonGuardDuty -> b4b3b0b27e29f5db632f1c5405f658e3 - UnmodifiableEntity: Cannot perform the operation on the protected role 'AWSServiceRoleForAmazonGuardDuty' - this role is only modifiable by AWS
`

### PR DESCRIPTION
**Pull request by svenwltr**
> Fixes #331

Additionally, it adds all properties to IAMRolePolicy resources.

@rebuy-de/prp-aws-nuke Please review.

### CODE REVIEW

File: resources/iam-role-policy.go [hunk 1]
```diff
@@ -65,10 +68,17 @@ func ListIAMRolePolicies(sess *session.Session) ([]Resource, error) {
 	return resources, nil
 }
 
+func (e *IAMRolePolicy) Filter() error {
+	if strings.HasPrefix(aws.StringValue(e.role.Path), "/aws-service-role/") {
+		return fmt.Errorf("cannot alter service roles")
```
Comment (bjoernhaeuser) [inline resources/iam-role-policy.go]: why is this an error?

Comment (svenwltr) [reply resources/iam-role-policy.go]: Back then I thought it would be smart to design the interface like this ...

Comment (bjoernhaeuser) [reply resources/iam-role-policy.go]: Understood. Though we could log here and still return `nil`.

Comment (svenwltr) [reply resources/iam-role-policy.go]: That would not work, because we want the control loop to know that it should not attempt to delete this resource.

**Pull request merged**